pom.xml

```diff
@@ -2503,9 +2503,6 @@
             <recompileMode>incremental</recompileMode>
             <useZincServer>true</useZincServer>
             <args>
-              <arg>-unchecked</arg>
```

Comment:
OK so the idea is we can't apply these to both 2.12 and 2.13? why not `unchecked` though?

Reply:
With `-unchecked`, Scala 2.13 has compilation error as follow:
```
[ERROR] /spark-source/sql/core/src/test/scala/org/apache/spark/sql/streaming/sources/StreamingDataSourceV2Suite.scala:293: a pure expression does nothing in statement position
```

So Scala 2.12 use  `-unchecked`, Scala 2.13 use `-Wconf:cat=unchecked:wv`

Reply:
Address be95a1b fix this